test/stdlib/Inputs/CommonArrayTests.gyb

```diff
@@ -572,6 +572,11 @@ ${Suite}.test("${ArrayType}/init(unsafeUninitializedCapacity:...:)/validCount")
 }
 
 ${Suite}.test("${ArrayType}/init(unsafeUninitializedCapacity:...:)/reassignBuffer") {
+  guard #available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *) else {
+    // When back-deployed to 5.0, this coding error does not get detected.
```

Comment:
cc @natecook1000

Reply:
That's… odd. Did I add that check in something that's now part of the ABI? LGTM in any case, thx @lorentey!

Reply:
The buffer reassign check wasn't in 5.0's underscored initializer, so code using the new public initializer won't get this check when it's deployed on macOS 10.14.2 / iOS 12.2 etc. I don't think this is a problem; the missing check won't break any working code.

Our ABI back deployment tests detected this change in behavior between the 5.0 and 5.1 stdlibs; the availability guard I added here acknowledges that this was an expected change.